python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -2031,25 +2032,67 @@ def _remeasure_star_background(self, result, background_to_photometric_ratio=Non
             self.log.debug("detected_fraction_orig = %.3f  detected_fraction_dilated = %.3f",
                            detected_fraction_orig, detected_fraction_dilated)
 
+        bbox = result.exposure.getBBox()
+
+        # Initialize the parameters for the pedestal iteration.
         pedestalBackgroundConfig = lsst.meas.algorithms.SubtractBackgroundConfig()
         for maskName in bgIgnoreMasksToAdd:
             if (maskName in result.exposure.mask.getMaskPlaneDict().keys()
                     and maskName not in pedestalBackgroundConfig.ignoredPixelMask):
                 pedestalBackgroundConfig.ignoredPixelMask += [maskName]
         pedestalBackgroundConfig.statisticsProperty = "MEDIAN"
         pedestalBackgroundConfig.approxOrderX = 0
-        pedestalBackgroundConfig.binSize = 64
-        pedestalBackgroundTask = lsst.meas.algorithms.SubtractBackgroundTask(config=pedestalBackgroundConfig)
-        pedestalBackgroundList = pedestalBackgroundTask.run(
-            exposure=result.exposure,
-            background=result.background,
-            backgroundToPhotometricRatio=background_to_photometric_ratio,
-        ).background
-        # Isolate the final pedestal background to log the computed value.
-        pedestalBackground = afwMath.BackgroundList()
-        pedestalBackground.append(pedestalBackgroundList[1])
-        pedestalBackgroundLevel = pedestalBackground.getImage().array[0, 0]
-        self.log.info("Subtracted pedestalBackgroundLevel = %.4f", pedestalBackgroundLevel)
+        pedestalBackgroundConfig.binSize = min(32, int(math.ceil(max(bbox.width, bbox.height)/4.0)))
+        self.log.info("Initial pedestal binSize = %d pixels", pedestalBackgroundConfig.binSize)
+        inPedestalIteration = True
+        cumulativePedestalLevel = 0.0
+        relativeStoppingCriterion = 0.05  # Relative change in cumulative pedestal value.
```

Comment:
This should have a longer comment (above it): "When the cumulative pedestal value changes by less than 5% from one bin size to the next we assume we are converged"